lib/auth/machineid/machineidv1/bot_instance_service.go

```diff
@@ -33,6 +34,18 @@ import (
 	"github.com/gravitational/teleport/lib/services"
 )
 
+const (
+	// AuthenticationHistoryLimit is the maximum number of authentication
+	// records to be recorded in a bot instance's .Status.LatestAuthentications
+	// field.
+	AuthenticationHistoryLimit = 10
+
+	// ExpiryMargin is the duration added to bot instance expiration times to
+	// ensure the instance remains accessible until shortly after the last
+	// issued certificate expires.
+	ExpiryMargin = time.Minute * 5
```

Comment:
I'm curious if there's any thoughts on the expiry margin. I imagine we want some nonzero buffer to account for any clock weirdness and so renewals don't fail because the bot instance somehow got pruned before a cert expires. I've just guessed at 5 minutes being good enough for this purpose.

That said, is there any value in keeping bot instances around longer? Do they provide worthwhile informational value? Or is it more valuable for a bot instance to expire shortly after a bot can no longer renew, so `tctl get bot_instance/foo` shows an approximately correct list of active bots?

Reply:
So from a technical angle, I think about 5 minutes makes sense to account for any "weirdness".

From a product angle, I think that for "historical data" we already have an audit log and this is probably what we ought to leverage for "I want to see info about things that happened in the past" rather than "show me the active state of my cluster". We probably ought to make sure long-term that anything that's accessible via the BotInstance is also accessible via the audit log. I do think it's quite nice w/ the current design that `tctl get bot_instance` shows anything thats currently capable of authenticating.

Reply:
Makes sense to me, I'll keep it at 5 minutes (and make a note to ensure audit events include an actionable amount of info on the bot instance, since they'll be acting as the historical record)